test/NuGet.Clients.Tests/NuGet.CommandLine.Test/MSBuildUtilityTest.cs

```diff
@@ -231,6 +232,51 @@ public void TestGetMsbuildDirectoryFromPATHENV()
             }
         }
 
+        [Theory]
+        [InlineData('"')]
+        [InlineData('<')]
+        [InlineData('>')]
+        [InlineData('|')]
+        public void GetMsbuildDirectoryFromPath_PATHENVWithInvalidChars_Succeeds(char invalidChar)
+        {
+            using (var vsPath = TestDirectory.Create())
+            {
+
+                if (RuntimeEnvironmentHelper.IsMono)
+                { // Mono does not have SxS installations so it's not relevant to get msbuild from the path.
+                    return;
+                }
+
+                var msBuild159BinPath = Directory.CreateDirectory(Path.Combine(vsPath, "MSBuild", "15.9", "Bin")).FullName;
+
+                var msBuild159ExePath = Path.Combine(msBuild159BinPath, "msbuild.exe").ToString();
+
+                using (var fs15 = File.CreateText(msBuild159ExePath))
+                {
+                    fs15.Write("foo 15.9");
+                }
+
+                var pathValue = Environment.GetEnvironmentVariable("PATH");
+
+                var newPathValue = new StringBuilder();
+                newPathValue.Append(invalidChar);
+                newPathValue.Append(msBuild159BinPath);
+                newPathValue.Append(invalidChar);
+                newPathValue.Append(';');
+                newPathValue.Append(pathValue);
+                
+                Environment.SetEnvironmentVariable("PATH", newPathValue.ToString());
+
+                // Act;
+                var toolset = MsBuildUtility.GetMsBuildToolset(userVersion: null, console: null);
+                Environment.SetEnvironmentVariable("PATH", pathValue);
```

Comment:
I think this can be improved. 
A few options that I can think of: 
1. Use IEnvironmentVariableReader in MSBUildAPiUtility, to allow for better mocking. 
1. Use try finally for the reseting of the env variable.